lib/pbench/agent/tool_meister.py

```diff
@@ -1316,121 +1316,120 @@ def _send_directory(self, directory, uri, ctx):
-            with o_file.open("w") as ofp, e_file.open("w") as efp:
-                cp = subprocess.run(
-                    [
-                        self.tar_path,
-                        "--create",
-                        "--xz",
-                        "--force-local",
-                        f"--file={tar_file}",
-                        target_dir,
-                    ],
-                    cwd=parent_dir,
-                    stdin=None,
-                    stdout=ofp,
-                    stderr=efp,
-                )
+            cp = subprocess.run(
+                tar_args,
+                cwd=parent_dir,
+                stdin=None,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
         except Exception:
             self.logger.exception("Failed to create tar ball '%s'", tar_file)
             failures += 1
         else:
             try:
                 if cp.returncode != 0:
-                    self.logger.error(
-                        "Failed to create tar ball; return code: %d", cp.returncode,
+                    self.logger.warning(
+                        "Failed to create tar ball first time; Error: %s, re-trying with --warning=none",
+                        str(cp.stdout),
                     )
-                    failures += 1
-                else:
+                    # record the tar error we encountered first time
+                    with open(target_dir / "tar_command_error") as file:
+                        file.write(str(cp.stdout))
+                    tar_args.insert(2, "--warning=none")
                     try:
-                        (_, tar_md5) = md5sum(tar_file)
+                        cp = subprocess.run(
+                            tar_args,
+                            cwd=parent_dir,
+                            stdin=None,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.STDOUT,
+                        )
+                        assert (
+                            cp.returncode == 0
+                        ), "Failed to create tar file second time"
                     except Exception:
-                        self.logger.exception("Failed to read tar ball, '%s'", tar_file)
+                        self.logger.exception(
+                            "Failed to create tar ball second time'%s'", tar_file
+                        )
                         failures += 1
-                    else:
-                        try:
-                            o_file.unlink()
-                        except Exception as exc:
-                            self.logger.warning(
-                                "Failure removing tar command output file, %s: %s",
-                                o_file,
-                                exc,
-                            )
+                try:
+                    (_, tar_md5) = md5sum(tar_file)
```

Comment:
Yeah it will hang, so if we failed the tar creation on the 2nd attempt, Peter suggested me that we at least need to send an empty tar file so that it wont hang.

Reply:
We need to make sure that we produce a file which we can successfully hash so that we end up performing the send....

Reply:
Of course if the drive is full, even an "empty" tar file takes up space and might not succeed. In which case we need to be prepared for failure.

I'd feel better about this path if we could say "we made two attempts and it didn't work, so tell the TDS control channel that we can't generate an artifact and skip the `PUT` entirely". (If we can't even do *that*, there's no hope.)